src/verify-request/verify-token.ts

```diff
@@ -29,7 +29,7 @@ export function verifyToken(routes: Routes, accessMode: AccessMode = DEFAULT_ACC
         try {
           // make a request to make sure oauth has succeeded, retry otherwise
           const client = new Shopify.Clients.Rest(session.shop, session.accessToken)
-          await client.get({ path: "metafields", query: {'limit': 1} }) 
+          await client.get({path: 'shop'});
```

Comment:
Hey @snehapawar9910 

> Just trying to understand if it would cause any latency issues as I've seen metafields API timing out during auth process multiple times.

I don't think you should see any of these issues with this change. The response body should just be an object with your shop name [and other details] - Paulo and I agree this is the cheapest call we can make here. As I mentioned above I am not a huge fan of making a request for data to check on the session, but, I wanted to unblock those using this library from not being able to publish.

Reply:
@JaKXz any reason to why the `metafields` REST endpoint is now deprecated? From my understanding, as of `2022-04`, this REST endpoint still exists and nothing looks to be deprecated: https://shopify.dev/api/admin-rest/2022-04/resources/metafield#get-metafields

Is it maybe because this code calls into an older API version (or it's calling the un-versioned endpoint), and the deprecation you are mentioning is the `value_type` deprecation? The reason I ask is that our app still relies on the call below, and I'd prefer to keep relying on the `metafields` resource and version it to `2022-01` rather than switch over to the `shop` resource.

Some inputs would be great here... :)


```
const response = await fetch(
  `https://${session.shop}/admin/metafields.json`,
  {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'X-Shopify-Access-Token': session.accessToken,
    },
  },
);
```